src/data/matrix/basic.lean

```diff
@@ -1083,6 +1201,19 @@ def reindex (eₘ : m ≃ l) (eₙ : n ≃ o) : matrix m n α ≃ matrix l o α
   left_inv  := λ M, by simp,
   right_inv := λ M, by simp, }
 
+def reindex_prod_assoc :
+matrix ((l × m) × n) ((o × p) × q) α ≃ matrix (l × m × n) (o × p × q) α :=
+reindex (equiv.prod_assoc _ _ _) (equiv.prod_assoc _ _ _)
+
+def reindex_prod_comm_snd : matrix l (m × n) α ≃ matrix l (n × m) α :=
+reindex (equiv.refl _) (equiv.prod_comm _ _)
+
+def reindex_prod_comm_fst : matrix (l × m) n α ≃ matrix (m × l) n α :=
+reindex (equiv.prod_comm _ _) (equiv.refl _)
+
+def reindex_prod_comm : matrix (m × n) (p × q) α ≃ matrix (n × m) (q × p) α :=
+reindex (equiv.prod_comm _ _) (equiv.prod_comm _ _)
```

Comment:
This overlaps with #8147 - I think it would be best to leave these out of this PR for now, but I'll drop a reference to these lines on that PR so we don't forget about them.

Reply:
Hi, Eric. I had a look at that one. The overlap this just the first simple lemma. I think that PR's main result is a linear equivalence (I remember the author of that one is on vacation). As this is a minor conflict, how about keeping them in this PR if you think they fit the file?

Reply:
In #8147 we decided they did not really belong in this file - I think it's best to leave these out until #8147 is resolved. I added https://github.com/leanprover-community/mathlib/pull/8147#issuecomment-879806163 so that we don't forget about these.

Reply:
OK, cool!